docs/README.md

```diff
@@ -263,24 +264,113 @@ Structure the `search-result-layout` or the `search-result-layout.customQuery`,
+| :---------: | :------: | :---------------------------------------------------------------------: | :-------------: |
+| `name`   | `string` | ![https://img.shields.io/badge/-Mandatory-red](https://img.shields.io/badge/-Mandatory-red) Layout name. This value must be unique i.e. not equal to other layout names declared in the `gallery` block. | `undefined` |
+| `component`   | `string` | ![https://img.shields.io/badge/-Mandatory-red](https://img.shields.io/badge/-Mandatory-red) Name of the [Slot](https://vtex.io/docs/concepts/slots) that will declare the block to be rendered in this layout. The same Slot name used as this prop's value must be declared afterwards as a prop (in the `gallery` block) whose value must be the desired block name. Caution: the `component`  prop's value must be equal to the Slot name responsible for declaring the desired block, thus being PascalCased, not to the desired block name itself. Check out the example below in order to understand the underlying logic behind this prop. | `undefined` |
+| `itemsPerRow`   | `number` / `object` | ![https://img.shields.io/badge/-Mandatory-red](https://img.shields.io/badge/-Mandatory-red) Number of items to be displayed in each row of this layout. This prop works with [responsive values](https://vtex.io/docs/app/vtex.responsive-values/), therefore it also accepts an object with different numbers for desktop, tablet or phone screen sizes (*see the table below*). | `undefined` |
+
+- **`itemsPerRow` object:** 
+  
+| Prop name | Type     | Description | Default value   |
+| :-------: | :------: | :--------:  | :-------------: | 
+| `desktop` | `number` | Number of slides to be shown on desktop devices. |  `undefined` | 
+| `tablet` | `number` | Number of slides to be shown on tablet devices. | `undefined` | 
+| `phone` | `number` |  Number of slides to be shown on phone devices.   | `undefined` | 
+
+For example:
+
+```json
+{
+  "gallery": {
+    "props": {
+      "layouts": [
+        {
+          "name": "whole",
+          "component": "WholeLineSummary",
+          "itemsPerRow": 1
+        },
+        {
+          "name": "many",
+          "component": "ManyByLineSummary",
+          "itemsPerRow": {
+            "desktop": 5,
+            "mobile": 2
+          }
+        }
+      ],
+      "WholeLineSummary": "product-summary.shelf#whole",
+      "ManyByLineSummary": "product-summary.shelf#many"
+    }
+  }
+}
+```
+
+#### `gallery-layout-switcher` block
+
+Allows users to shift between the available `gallery`'s layouts. 
 
-Therefore, don't forget to check out the [Flex Layout documentation](https://vtex.io/docs/apps/layout-blocks/vtex.flex-layout@0.14.0) for more on how to configure your search results page. 
+It receives no props and expects as child the `gallery-layout-option` block described below.
 
-Below you can find the existing props for each of the blocks, in addition to the the rules that govern them. 
+#### `gallery-layout-option` block
```

Comment:
We should add this block to the app's block list (at the beginning of the doc).

Reply:
I didn't fully understand this block functionality, therefore I couldn't properly understand as well its props.  What does it do on the UI?

Reply:
This block is the button used to switch to a given layout. In the description of this PR, the GIF shows me clicking on it to navigate between layouts.